os/fs/smartfs/smartfs_smart.c

```diff
@@ -1738,3 +1738,17 @@ static int smartfs_stat(struct inode *mountpt, const char *relpath, struct stat
 /****************************************************************************
  * Public Functions
  ****************************************************************************/
+
+/****************************************************************************
+ * Name: get_smartfs_optimal_read_size
+ *
+ * Description: Return the optimal buffer size for reading from SmartFs to
+ *   avoid repetitive reading of sectors. This size is the number of actual
+ *   data bytes in the sector
+ *
+ * **************************************************************************/
+
+int get_smartfs_optimal_read_size()
+{
+	return (CONFIG_MTD_SMART_SECTOR_SIZE - sizeof(struct smartfs_chain_header_s) - get_smart_mtd_header_size());
```

Comment:
block driver layer. should be hided to fs layer, this can' be accepted. you should use ioctl

Reply:
@amogh-samsung With the loadable configuration, this API can't be called from applications unless a new ioctl interface is introduced.

Reply:
Updated